Loading multiple teams per folder

mariohmol opened the pull request:
Feature Request

As proposed in this ticket https://github.com/SongChiYoung/autogen-oaiapi/issues/32

This PR introduce the ability to load multiple TeamManager agents via folder, giving it support to be called in OpenAI REST style.

Using the model body data is possible to switch which agent to ask

mariohmol committed:
Loading multiple teams per folder

SongChiYoung commented:
Hey @mariohmol 
thanks so much for the contribution — this is an awesome idea and a super helpful feature!

I just have two small requests before we merge:
	1.	Example Folder: I loved the example you added. But I noticed there’s no examples/agents folder yet. Could you include that so it becomes a fully runnable example out of the box? I believe examples should work with just python exmaple_....py without any extra setup.
	2.	agent_manager.py Location: I’m a bit unsure about placing agent_manager.py directly at the root level. My plan was to organize various managers under something like a manager/actors/ or manager/agents/ folder (feel free to suggest a better name if you have one!). Would you mind moving it accordingly? Just want to keep the structure clean and avoid confusion later on.

Again, huge thanks for your contribution — I really appreciate it (and love you for it 😄).
Looking forward to your update!

mariohmol commented:
I'm sorry, made it so late night that didn't make a proper CR. Let me work on those

mariohmol committed:
CR review fixes, better comments and fully team working example

mariohmol committed:
Moving Agent manager to his own folder

mariohmol commented:
Allright, here we go

* removed some dummy comments
* added useful comments
* added an fully example, with a team.json and a comment explaning how it works
* moved the agentmanager to the manager folder

Lmk if you invision any other change!

thanks

mariohmol committed:
adding fixes for supporting stream requests, and variation of content (can be object with type and text)

SongChiYoung commented:
Cool, I think we have other changes like them
- adding this for register routine
- adding api key routine
- adding it for model api

however, we could dev them with other PR(issue)s what about your opinions?

SongChiYoung commented:
@mariohmol 
Cool I will merge it. thanks to your contribute.
